tf_agents/environments/py_environment.py

```diff
@@ -168,7 +168,7 @@ def step(self, action):
         observation: A NumPy array, or a nested dict, list or tuple of arrays
           corresponding to `observation_spec()`.
     """
-    if self._current_time_step is None:
+    if self._current_time_step is None or self._current_time_step.is_last():
```

Comment:
One challenge here is that python environments could be batched. So each call to step would be stepping each of the sub environments, and it is possible that only some of them have current_time_step.is_last() == True.

Reply:
Ah yes, I didn't think about batched environments, thanks. Mmh, not sure what's the best option here: we could reset if the environment is not batched, which would probably take care of batched environments as well, indirectly (since they are probably made of multiple non batched environments, which will get a call to `step()` and reset if needed. However, I'm not sure what the implications are.

Reply:
Good point, yeah I think that should handle the batched environments. 

If we go ahead with this change, we should also make similar changes in the TF environment so that things are consistent, 
https://github.com/tensorflow/agents/blob/master/tf_agents/environments/tf_environment.py

And also in the different suites. Maybe just this file I think:
https://github.com/tensorflow/agents/blob/58c7a5fdb01359ffb2cefcda9350aa65a0907eb5/tf_agents/environments/gym_wrapper.py#L153